geo/src/test/java/com/tomtom/speedtools/geometry/GeoPolyLineTest.java

```diff
@@ -74,6 +74,26 @@ public void testAsLines() {
     }
 
     @Test
+    public void testElevation() {
+        LOG.info("testElevation");
+        final GeoPoint x1 = new GeoPoint(0.0, 0.0);
+        final GeoPoint x2 = new GeoPoint(1.0, 0.0, 100.0);
+        final GeoPoint x3 = new GeoPoint(2.0, 0.0);
+        final GeoPoint x4 = new GeoPoint(3.0, 0.0, 200.0);
+        final GeoPoint x5 = new GeoPoint(4.0, 0.0, 300.0);
+        final GeoPoint x6 = new GeoPoint(5.0, 0.0);
+        final List<GeoPoint> p1 = Lists.asList(x1, new GeoPoint[]{x2, x3, x4, x5, x6});
+        final GeoPolyLine poly1 = new GeoPolyLine(p1);
+        Assert.assertEquals(100.0, poly1.get(0).getElevationMetersOrNaN(), 0.0000001);
+        Assert.assertEquals(100.0, poly1.get(1).getElevationMetersOrNaN(), 0.0000001);
+        Assert.assertEquals(100.0, poly1.get(2).getElevationMetersOrNaN(), 0.0000001);
+        Assert.assertEquals(200.0, poly1.get(3).getElevationMetersOrNaN(), 0.0000001);
+        Assert.assertEquals(300.0, poly1.get(4).getElevationMetersOrNaN(), 0.0000001);
+        Assert.assertEquals(300.0, poly1.get(5).getElevationMetersOrNaN(), 0.0000001);
```

Comment:
An improvement could be to use a DELTA constant, like in the GeoTest.

Reply:
Done